### ISSUE TITLE
EntityLog#registerCreate() fails to register a BaseDbGeneratedIdEntity

### ISSUE THREAD

**Issue by pavel-grigorev** [cuba-platform/cuba#2129]
### Environment

- Platform version: 6.10.9
- Client type: Web
- Database: MySql as an additional data store
- Operating system: macOS

### Description of the bug or enhancement

- Minimal reproducible example

Given an entity:
```java
@AttributeOverrides({
    @AttributeOverride(name = "id", column = @Column(name = "segment_id"))
})
@Table(name = "segment")
@Entity(name = "proj$Segment")
public class Segment extends BaseIdentityIdEntity {
}
```

When the entity is created by calling Metadata#create() and then committed by calling DataManager#commit() the following exception is thrown
```
11:14:27.018 WARN c.h.cuba.security.app.EntityLog - Unable to log entity com.proj.entity.Segment-?(89000f2e-d490-4f45-b9ab-c47a24ebaa70) [new,managed], id=?(89000f2e-d490-4f45-b9ab-c47a24ebaa70)
java.lang.IllegalStateException: Cannot get primary key value: entity is null
at com.haulmont.cuba.core.entity.IdProxy.getNN(IdProxy.java:135)
at com.haulmont.cuba.security.app.EntityLog$1.afterCommit(EntityLog.java:210)
at org.springframework.transaction.support.TransactionSynchronizationUtils.invokeAfterCommit(TransactionSynchronizationUtils.java:133)
at org.springframework.transaction.support.TransactionSynchronizationUtils.triggerAfterCommit(TransactionSynchronizationUtils.java:121)
at org.springframework.transaction.support.AbstractPlatformTransactionManager.triggerAfterCommit(AbstractPlatformTransactionManager.java:962)
at org.springframework.transaction.support.AbstractPlatformTransactionManager.processCommit(AbstractPlatformTransactionManager.java:807)
at org.springframework.transaction.support.AbstractPlatformTransactionManager.commit(AbstractPlatformTransactionManager.java:734)
at com.haulmont.cuba.core.sys.TransactionImpl.commit(TransactionImpl.java:104)
at com.haulmont.cuba.security.app.EntityLog.registerCreate(EntityLog.java:373)
```

- Caused by

A line from the afterCommit method:
```
Number id = item.getDbGeneratedIdEntity().getId().getNN();
```

getId() returns a copy of the IdProxy instance with its entity nulled out which in turn causes getNN() to throw IllegalStateException.

**Comment by pavel-grigorev** [comment]
Is this going to be fixed in 6.10?

**Comment by andreysubbotin** [comment]
7.0+. It's hard to merge changes in the 6.10 release.
